File: Technology Platforms (Java, .NET)/Java (lab 1, 2, 3 - multithreading, recursive class)/src/main/java/org/example/data/PersonGenerator.java
package org.example.data;

import org.example.Person;

import java.util.*;
import java.util.concurrent.BlockingQueue;
import java.util.concurrent.LinkedBlockingQueue;

public class PersonGenerator {

    private static final String[] IMIONA = {"Jan", "Anna", "Piotr", "Maria", "Krzysztof", "Agnieszka", "Czesław", "Miłosz",
            "Mirosław", "Wacław", "Zbigniew", "Mariusz", "Michał", "Dawid", "Antoni", "Robert"};
    private static final String[] NAZWISKA = {"Kowalski", "Nowak", "Wiśniewski", "Dąbrowski", "Lewandowski", "Pazdan", "Milik", "Glik", "Krychowiak", "Szczęsny"};
    private static final Random RANDOM = new Random();

    public static Person generateRandomPerson() {
        String imie = IMIONA[RANDOM.nextInt(IMIONA.length)];
        String nazwisko = NAZWISKA[RANDOM.nextInt(NAZWISKA.length)];
        int wiek = RANDOM.nextInt(80) + 1;
        HashSet<Person> dzieci = new HashSet<>();
        if(wiek > 20) {
            int ilosc_dzieci = RANDOM.nextInt(2);
            for(int i = 0; i < ilosc_dzieci; i++) {
                String imie_dziecka = IMIONA[RANDOM.nextInt(IMIONA.length)];
                int wiek_dziecka = wiek-20;
                dzieci.add(new Person(imie_dziecka, nazwisko, wiek_dziecka));
            }
        }
        return new Person(imie, nazwisko, wiek, dzieci);
    }

    public static List<Person> generatePeopleList(int count) {
        List<Person> peopleList = new ArrayList<>(count);
        for (int i = 0; i < count; i++) {
            peopleList.add(generateRandomPerson());
        }
        return peopleList;
    }

}

File: Technology Platforms (Java, .NET)/.NET (recursive class, LINQ, XML, and XPath, datagrid, multithreading)/PersonGenerator.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace WpfApp
{
    internal class PersonGenerator
    {
        private static int _nextId = 1;
        private static readonly string[] NAMES = { "Jan", "Anna", "Piotr", "Maria", "Krzysztof", "Agnieszka", "Czesław", "Miłosz",
            "Mirosław", "Wacław", "Zbigniew", "Mariusz", "Michał", "Dawid", "Antoni", "Robert", "Anastazja", "Barbara", "Katarzyna", "Maja", "Zuzanna", "Oliwia", "Julia" };
        private static readonly string[] SURNAMES = { "Kowal", "Nowak", "Maj", "Mazur", "Mucha", "Pazdan", "Milik", "Glik", "Krychowiak", "Muller", "Lato", "Lis", "Kruk", "Kowalik", "Bednarz" };
        private static readonly Random RANDOM = new Random();

        public static Person GenerateRandomPerson()
        {
            string name = NAMES[RANDOM.Next(NAMES.Length)];
            string surname = SURNAMES[RANDOM.Next(SURNAMES.Length)];
            int age = RANDOM.Next(1, 91);
            int id = _nextId++;

            Person person = new Person(id, name, surname, age);
            AdvancedObservableCollection<Child> children = new AdvancedObservableCollection<Child>();
            if (person.Details.Category != PersonCategory.Dziecko && person.Details.Category != PersonCategory.Nastolatek)
            {
                int iloscDzieci = RANDOM.Next(3);
                for (int i = 0; i < iloscDzieci; i++)
                {
                    string childName = NAMES[RANDOM.Next(NAMES.Length)];
                    int minChildAge = Math.Max(1, age - 50);
                    int maxChildAge = Math.Max(1, age - 20);
                    int childAge = RANDOM.Next(minChildAge, maxChildAge + 1);
                    children.Add(new Child(childName, childAge));
                }
            }

            person.Children = children;
            return person;
        }

        public static List<Person> GeneratePeopleList(int count)
        {
            List<Person> peopleList = new List<Person>(count);
            for (int i = 0; i < count; i++)
            {
                peopleList.Add(GenerateRandomPerson());
            }
            return peopleList;
        }

        public static Child GenerateRandomChild()
        {
            string name = NAMES[RANDOM.Next(NAMES.Length)];
            int age = RANDOM.Next(1, 18);
            return new Child(name, age);
        }

        public static int GetNextId()
        {
            return _nextId++;
        }

        public static void ReduceId()
        {
            _nextId--;
        }

        public static void ResetId()
        {
            _nextId = 1;
        }
    }
}
